src/Pages/Applications/DevtronApps/service.types.ts

```diff
@@ -55,3 +56,77 @@ export interface ResourceConfig {
 export interface EnvConfigDTO {
     resourceConfig: ResourceConfig[]
 }
+
+export enum AppEnvDeploymentConfigType {
+    PUBLISHED_ONLY = 'PublishedOnly',
+    DRAFT_ONLY = 'DraftOnly',
+    PUBLISHED_WITH_DRAFT = 'PublishedWithDraft',
+    PREVIOUS_DEPLOYMENTS = 'PreviousDeployments',
+    DEFAULT_VERSION = 'DefaultVersion',
+}
+
+export interface DraftMetadata {
+    appId: number
+    envId: number
+    resource: number
+    resourceName: string
+    action: number
+    data: string
+    userComment: string
+    changeProposed: boolean
+    protectNotificationConfig: { [key: string]: null }
+    draftId: number
+    draftVersionId: number
+    draftState: number
+    approvers: string[]
+    canApprove: boolean
+    commentsCount: number
+    dataEncrypted: boolean
+    isAppAdmin: boolean
+}
+
+export interface ConfigMapSecretDataConfigDatumDTO extends ConfigDatum {
```

Comment:
I think there is some typo in ConfigDatum

Reply:
it's valid

Response by ChatGPT

```
The term “ConfigDatum” combines “Config” (short for “Configuration”) and “Datum” (singular of “Data”). While it’s not a standard English word, it’s a valid construction in the context of programming and technical language, particularly if you’re referring to a singular piece of configuration data.

	•	Config: Abbreviation for “configuration.”
	•	Datum: The singular form of “data,” used to refer to a single piece of information.

So, “ConfigDatum” would logically refer to a single configuration item or piece of configuration data. If you prefer more conventional terminology, you could use “ConfigurationItem” or “ConfigEntry.”
```

Reply:
Noice